### PR TITLE
Cleanup, add VS Code auto formatting, reformat all the code and use absolute paths

### PR DESCRIPTION
**Pull request by RobbieZhuang**
# VS Code Auto Formatting and Absolute Paths 🥳 🥳 
**The Run Down**
---

**Auto Formatting**: We shouldn't have to **worry-about-formatting**™ our code properly!! This PR brings a new VSCode workspace file `sesoc.code-workspace` which automatically formats our code upon saving a file! CI will still continue to work as usual to catch any formatting or linting issues.

**Absolute Paths**: All of our paths use to be relative but I've swapped them all over to be absolute in relation to the root of the source code (i.e. `src`). This will help us with more readable and clean imports in the future!

**Note**: There are no changes to our actual website.

**Changelog**
---
This change also introduces a few new formatter rules and an update to the `package-lock.json` file (hence the huge loc change).

* Updated react-scripts to version 4.0.2 in order to support absolute paths
* Updated all imports to use absolute paths over original relative paths for clarity
   * Introduced `jsconfig.json` in the root of the directory to facilitate absolute paths
* Included `sesoc.code-workspace` which enables format-on-save
* Removed formatting rules from .eslintrc.js and made compatible with prettier
   * Introduced `.prettierrc` to add formatter preferences and `.prettierignore` for ignored files 
* Remove unused firebase config files
* Removed Platforms component and a dependency on a `yarn.lock` and some other modules
* Reformatted a lot of code while keeping CI happy

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,65 +1,29 @@
...
-            "pragma": "React",
-            "version": "detect",
-            "flowVersion": "0.53"
-        },
-        "linkComponents": [
-            {"name": "Link", "linkAttribute": "to"},
-            {"name": "OutboundLink", "linkAttribute": "to"}
-        ]
-    },
-    rules: {
```
Comment (RobbieZhuang) [inline .eslintrc.js:23]: We will let prettier take care of formatting rules - thus remove all of them from this file

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -16,15 +16,15 @@ jobs:
         node-version: [12.x]
 
     steps:
-    - uses: actions/checkout@v2
```
Comment (RobbieZhuang) [inline .github/workflows/build.yml:19]: These are just auto-formatted with prettier

File: .gitignore [hunk 3]
```diff
@@ -14,7 +14,7 @@
 #firebase.json
 
 # misc
-.DS_Store
+**/.DS_Store
```
Comment (RobbieZhuang) [inline .gitignore:17]: ignore all the .DS_Stores!!!

File: jsconfig.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+  "compileOnSave": false,
```
Comment (RobbieZhuang) [inline jsconfig.json:2]: More static code checks (not related to prettier)

File: jsconfig.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  "compilerOptions": {
+    "baseUrl": "src",
```
Comment (RobbieZhuang) [inline jsconfig.json:4]: Setting baseUrl to src allows us to use absolute paths with src as the root

File: package.json [hunk 6]
```diff
@@ -1,44 +1,42 @@
 {
-    "private": true,
```
Comment (RobbieZhuang) [inline package.json:2]: Autoformatter

File: sesoc.code-workspace [hunk 7]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (RobbieZhuang) [inline sesoc.code-workspace:1]: When editing, make sure to jump into this workspace configuration in order to get noice features like auto formatting when saving files!!

**Review by alexgirard — APPROVED** [review]
lol that is a lot of formatting changes

**Comment by RobbieZhuang** [comment]
THANK U ALEX

**Pull request merged**